### DISCUSSION TITLE
Which plugin to use for text box entry?

### DISCUSSION THREAD

**Discussion by kevinvschie** [jspsych/jsPsych#1083] [Q&A]
I'm trying to set a memory test for participants in jspsych. It's relatively simple. I want to present participants with a cue word (e.g., orchard) and they have to type in the response word (e.g., apple). A few things I want or need to put in there:

- I want to present cues one at a time 
- I have a lot of cue words, so I probably need a timeline to cycle through all of them. (Are all plugins compatible with this?)
- I would like to code accuracy (if possible). 
- Ideally I would like to have a text box where participants can see what they type (either below or right next to the cue)
- Hitting enter makes them proceed to the next cue word. 

I could not readily find what plugin would achieve this. Any ideas?

**Comment by kevinvschie** [comment]
Okay, thus far I figured out I could do almost all of this with the survey-text plugin. I only need some advice on coding for accuracy.

**Comment by melisa-akan** [comment]
Hi @kevinvschie, 

I also collected response from users and checked for accuracy using the survey-text plugin. The target word your subjects will input will be saved in "responses" as you can see under "Data Generate" on the [plugin page](https://www.jspsych.org/plugins/jspsych-survey-text/). Assuming that you will present one cue word per trial, the target word input by the user will be saved in: data.responses.Q0 . Q0 stands for Question 0.  The response is saved in JSON format. Here is what I did to access the response:  `JSON.parse(data.responses).Q0`

I converted both the correct target word and the word input by the user to capitals using the `.toUpperCase()` method so that the response would be counted as correct as longs as the letters were the same. I provide the relevant part of my code below in case it helps to see it in context. I wrote some comments for you in capital letters. I hope this helps. If not, happy to talk more. I am struggling a lot myself so I would love to help when I can.

(I highly recommend watching the [Session 3](https://www.youtube.com/watch?v=LP7o0iAALik&feature=youtu.be) of the [Moving Research Online workshop ](http://www.movingresearchonline.info/?fbclid=IwAR3c64P5uj2wp2XpSfHaoBWvJrg4lI5AdqRE4K7SHabYwUXct6OhuZkrVqo) if you haven't already. In that tutorial, Becky Gilbert covers, with much clarity, how to code for accuracy and provide feedback.)


`	
			var PreExpKnow_B2p_1 = {

			type: 'survey-text',
			questions: [
			{prompt: "<p> Please type in the name of this person and press Enter </p> <p> <img src=Faces/" + preExpFaces[i] +".jpg> </p>"}
			],
			data: {
				name: image_name, // THIS IS ANALOGOUS TO YOUR TARGET WORD (I defined this earlier in my code)
				cond:'know',
				trial_num:i+1,
				face: preExpFaces[i]
			},
			on_finish: function(data){
						var acc = false; //SET ACCURACY TO FALSE AND CHANGE IT TO CORRECT IF RESPONSE IS ACCURATE
				
						if (data.name.toUpperCase() == JSON.parse(data.responses).Q0.toUpperCase()){ //IF CORRECT TARGET WORD IS THE SAME WITH THE INPUT WORD, CHANGE ACCURACY TO TRUE
								acc = true;
							} 
							data.accuracy = acc;  // SAVE THIS IN THE DATA OBJECT
							console.log("acc: " + data.accuracy);
												
						} 
			} 
			timeline.push(PreExpKnow_B2p_1);
			`

**Comment by chrisbrickhouse** [comment]
This question has largely been answered, but for those who run into similar problems in the future, I have a partial solution to this in my experiment templates repo under [text-norming](https://github.com/chrisbrickhouse/Experiment-templates/tree/master/text-norming). You can place your list of stimuli in `wordlist.txt` and use `make stimuli` to update the experiment javascript. Then you can run the experiment and it will present participants with a random word from that list along with a text box for their response. You can change the textbox's placeholder text and other parts of the experiment by editing `semantic-associations-text.js`. You can change the demographic questions at the end by editing `demo-questions.js`
